webapp/src/app/tests/test-helper.ts

```diff
@@ -0,0 +1,33 @@
+import { MapLoaderService } from '../shared/map-loader.service';
+import { HttpClientService } from '../services/http-client.service';
+import { CrossCodeMap } from '../models/cross-code-map';
+import { BasePath, FileExtension, PathResolver } from '../shared/path-resolver';
+import { CCMap } from '../shared/phaser/tilemap/cc-map';
+import { Subscription } from 'rxjs';
+
+export class TestHelper {
+	public static async loadMap(service: MapLoaderService, http: HttpClientService, mapName: string): Promise<{ imported: CrossCodeMap, ccmap: CCMap }> {
+		return new Promise(((res) => {
+			let imported: CrossCodeMap;
+			let sub: Subscription;
+			// eslint-disable-next-line prefer-const
+			sub = service.tileMap.subscribe(map => {
+				if (!map) {
+					return;
+				}
+				
+				sub!.unsubscribe();
```

Comment:
There seems to be a some sort of race condition or test ordering issue here because one test failed because `sub` was undefined. Since I couldn't reprocude the issue I suggest making a note/issue and investigating later.